pkg/kubectl/cmd/macro/secret.go

```diff
@@ -0,0 +1,132 @@
+)
+
+// NewCmdCreateSecretOpaque is a command to create generic secrets from files, directories, or literal values
+func NewCmdCreateSecretOpaque(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "opaque NAME [--type=string] [--from-file=[key=]source] [--from-literal=key1=value1] [--dry-run=bool]",
+		Short:   "Create a secret from a local file, directory or literal value.",
+		Long:    secretLong,
+		Example: secretExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateSecretOpaque(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().String("generator", "", "The name of the API generator to use.  If not specified, default to secret/v1")
+	cmd.Flags().StringSlice("from-file", []string{}, "Key files can be specified using their file path, in which case a default name will be given to them, or optionally with a name and file path, in which case the given name will be used.  Specifying a directory will iterate each named file in the directory that is a valid secret key.")
+	cmd.Flags().StringSlice("from-literal", []string{}, "Specify a key and literal value to insert in secret (i.e. mykey=somevalue)")
+	cmd.Flags().String("type", "", "The type of secret to create")
+	cmd.Flags().Bool("dry-run", false, "If true, only print the object that would be sent, without sending it.")
+	return cmd
+}
+
+// CreateSecretOpaque is the implementation the create secret opaque command
+func CreateSecretOpaque(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, args []string) error {
+	stringSliceParams := []string{"from-file", "from-literal"}
+	return createCommandInternal(f, cmdOut, cmd, args, "secret/v1", commandParamStringSliceVistor(stringSliceParams))
+}
+
+const (
+	secretForDockerRegistryLong = `
+Create a new secret for use with Docker registries.
+
+Dockercfg secrets are used to authenticate against Docker registries.
+
+When using the Docker command line to push images, you can authenticate to a given registry by running
+  'docker login DOCKER_REGISTRY_SERVER --username=DOCKER_USER --password=DOCKER_PASSWORD --email=DOCKER_EMAIL'.
+That produces a ~/.dockercfg file that is used by subsequent 'docker push' and 'docker pull' commands to
+authenticate to the registry.
+
+When creating applications, you may have a Docker registry that requires authentication.  In order for the
+nodes to pull images on your behalf, they have to have the credentials.  You can provide this information
+by creating a dockercfg secret and attaching it to your service account.`
+
+	secretForDockerRegistryExample = `  // If you don't already have a .dockercfg file, you can create a dockercfg secret directly by using:
+  $ kubectl create secret docker-registry my-secret --docker-server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL`
```

Comment:
How many times can we squeeze "docker" into one commandline?  Do all the flags need to be prefixed?